app/client/components/delivery/address/deliveryAddressForm.tsx

```diff
@@ -175,12 +178,16 @@ const FormContainer = (props: FormContainerProps) => {
   const addressChangeInformation = Object.values(
     props.contactIdToArrayOfProductDetail
   )
-    .flatMap(babelFlatMapFunction)
-    .map(productDetail => {
-      const friendlyProductName = capitalize(
-        ProductTypes.contentSubscriptions.mapGroupedToSpecific?.(productDetail)
-          .friendlyName
-      );
+    .flatMap<ProductDetail>(flattenEquivalent)
+    .map(productDetail => ({
+      productDetail,
+      productType: ProductTypes.contentSubscriptions.mapGroupedToSpecific?.(
+        productDetail
+      )
+    }))
```

Comment:
calculating the `ProductType` here first and passing it on as an object of shape 
```javascript
{
  productDetail: ProductDetail,
  productType?: ProductType
}
```